js/src/workers/ComputeAndInjectDiffsWorker.js

```diff
@@ -89,13 +89,17 @@ function computeAndInjectDiffs({ previousData, currentData }) {
   const currentImageData = imageTo2DArray(
     currentData, maxWidth - currentData.width);
 
+  self.postMessage({ progress: 30 });
```

Comment:
If the diff takes disproportionately longer than the other operations as the image size goes up, it might be nice to come up with a quick formula to make these progress numbers more accurate.